verifier/services/reorg_detector_service.go

```diff
@@ -59,15 +59,14 @@ type ReorgDetectorConfig struct {
 // - Runs alongside SourceReaderService for each chain
 // - Uses same SourceReader instance to share RPC connections.
 type ReorgDetectorService struct {
-	services.StateMachine
+	sync   services.StateMachine
+	stopCh services.StopChan
+	wg     sync.WaitGroup
 
 	sourceReader chainaccess.SourceReader
-	headTracker  chainaccess.HeadTracker
 	config       ReorgDetectorConfig
 	lggr         logger.Logger
 	statusCh     chan protocol.ChainStatus
-	cancel       context.CancelFunc
```

Comment:
We use this pattern in many of the 1.7 services. I think getting rid of cancel might mean you need to re-evaluate how contexts are used throughout and add checks to `stopCh` wherever `ctx.Done()` is used